### ISSUE TITLE
Determine whether a build job will ever complete (is there a matching provisioner)

### ISSUE THREAD

**Issue by bpmct** [coder/coder#8247]
A workspace may be stuck in a "pending" state in a queue forever in the following cases:

- There are no provisioners
- There are no provisioners that match the tags required for the job (e.g. local provisioner or tagged provisioner)

Let's surface this to the user via the dashboard and CLI. Right now, the user will just see that their workspace is pending or in a queue, but it will never progress.

**Comment by Emyrk** [comment]
@sreya Should this be blocked by the provisioner refactor? I ask because currently we store provisioners in the DB, so this should not be tricky to implement. However there was a PR opened to maybe remove it from the DB as a precursor to refactoring the provisioner stuff.

I just do not want to implement ontop of something that will be removed.

**Comment by bpmct** [comment]
Fairly sure this is blocked by the provisioner refactor

**Comment by Emyrk** [comment]
Removed from the sprint until unblocked

**Comment by matifali** [comment]
Related and possible duplicate of #12331. At least we should have a way to delete such workspaces from the coder database. It could be similar to the orphan resources option for failed workspaces.

**Comment by johnstcn** [comment]
Needs RFC
